src/python/pants/backend/python/dependency_inference/module_mapper.py

```diff
@@ -410,11 +412,15 @@ def __post_init__(self) -> None:
 class PythonModuleOwnersRequest:
     module: str
     resolve: str | None
+    # If available and enabled resolve ambiguity by choosing the symbol provider with the
+    # closest common ancestor to this path. Must be a path relative to the build root.
+    locality: str | None = None
```

Comment:
If this is a path, can we call it so?

Reply:
yes it is, `locality_root`? (c.f. `source_root`..)

Reply:
I guess to me "locality" already implies that it's a path, so `locality_path` seems redundant?

Reply:
I have no mental schema in place for that word. Maybe there's just a better word?

Reply:
Been thinking about this, and could not come up with a succinct name.  So I think the obtuse, conceptual name might actually be good on balance because it forces you to read the comments to see what this means... 

Fortunately this is not part of the user-facing API, and anyone tinkering with dep inference is going to have to be pretty deeply in the weeds anyway. 

On the user-facing side, I have changed the option name and type, in case we need other granularities of locality in the future.